Concourse: add catkeys task to haiku-release pipeline

nielx opened the pull request:
This task creates the English catalogues for the current checkout, zips them up, and uploads them to the bucket with nightly images.

Currently, these catkeys are built per branch, per architecture. This will incur a slight overhead, as the translation tooling will only use the x86_gcc2h arch, but the size of these files are small so it is acceptable.

Only the latest catkeys build is retained. This gives the advantage of a predictable filename. There also is no need to keep other catkey archives.

Part of:
 * https://github.com/haiku/infrastructure/issues/37
 * https://dev.haiku-os.org/ticket/17530

nielx commented:
@kallisti5 : I tested the steps manual, but I did not manage to test the steps automatically. My machine does not have the resources to mirror the entire concourse setup. Is this something you could test?

kallisti5 commented:
Gave this one a try.  We need to ensure jam is in the general-worker container and cpio is as well before this one will work.  I also updated the change to upload in the root of an s3 bucket called haiku-translations

nielx commented:
I have updated the script and made three changes:
* I updated the bucket where things were copied to (though I had to guess/deduct, so please check).
* The dependency on cpio is now dropped; it uses cp which should be on the image.
* I have changed the worker to the toolchain worker, to add the availability of jam.

I hope this works around the issues without having to change the general worker. Could you, after applying any other changes you made, check whether this one works?

kallisti5 commented:
Seems to be working :-)

https://s3.wasabisys.com/haiku-translations/haiku-master-x86_64-catkeys.zip

kallisti5 commented:
@nielx if you confirm that zip looks good.. i'd say we can merge this one and close it

Also let me know if we need to do a build of r1beta3.  Assuming we might to help your automation?

nielx commented:
Contents looks good, sha256sum matches, so I'd say good to merge. Translations currently only track nightly, so no need to apply to r1beta3.
